docs/specifications/cva6_requirement_specification.adoc

```diff
@@ -0,0 +1,391 @@
+|===
+
+The user’s manual will detail the list of counters, events and related controls.
+
+=== Cache requirements  
+
+Caches increase the performance of the processor with regard to memory accesses. Most of their added value for the IP is specified through performance requirements in another section. Here below are specific requirements for these caches.
+
+The project would like to adopt the recently ratified [RVcmo] specification. The analysis yet needs to be performed and will likely lead to an evolution of this specification.
+
+==== L1 write-through data cache
+
+In the requirements below, L1WTD refers to the L1 write-through data cache that is part of the CVA6.
+
+The first two requirements express the write-through feature. Some requirements are useful for security- and safety-critical applications where a high level of timing predictability is needed.
+
+|===
+
+|L1W&#8209;10|L1WTD shall reflect all write accesses (stores) by the CVA6 core to the external memory after an upper-bounded number of cycles.
+|L1W&#8209;20|L1WTD shall not change the order of write accesses to the external memory with respect to the order of write accesses (stores) received from the CVA6 core.
+|L1W&#8209;30|L1WTD should offer the following size/ways configurations: +
+- 0 kbyte (no cache), +
+- 4 kbytes (4 or 8 ways), +
+- 8 kbytes (4, 8 or 16 ways), + 
+- 16 kbytes (4, 8 or 16 ways), +
+- 32 kbytes (8 or 16 ways).
+|L1W&#8209;40|L1WTD shall support datasize extension to store EDC, ECC or other information.
+|L1W&#8209;50|To interface with the P-Mesh coherence system of OpenPiton, L1WTD shall have a line invalidate external command that invalidates the content of a line upon request.
+|L1W&#8209;60|Some physical memory regions shall be configurable as not L1WTD cacheable at design time.
+|L1W&#8209;70|It shall be possible to invalidate L1WTD content with the `FENCE.T` command.
+|L1W&#8209;80|The replacement policy of L1WTD shall be LFSR (pseudo-random) or LRU (least recently used).
+|L1W&#8209;90|L1WTD should offer a feature to transform cache ways into a scratchpad.
+|L1W&#8209;100|A custom CSR shall allow to disable or enable L1WTD.
+|===
+
+Cache counters are defined in the performance counters.
+
+32 kbytes & 4 ways is not feasible with the current architecture. Other size/ways configurations may be implemented in the design.
+
+The design will support one replacement policy allowed by L1W-80.
+
+==== L1 Instruction cache
+
+In the requirements below, L1I refers to the L1 instruction cache that is part of the CVA6.
+
+Some requirements are useful for security- and safety-critical applications where a high level of timing predictability is needed.
+
+|===
+
+|L1I&#8209;10|L1I should offer the following size/ways configurations: +
+- 4 kbytes: 3, 4 or 8 ways, +
+- 8 kbytes: 4, 8, or 16 ways, +
+- 16 kbytes: 4, 8 or 16 ways, +
+- 32 kbytes: 8 or 16 ways.
+|L1I&#8209;20|L1I shall support datasize extension to store EDC, ECC or other information.
```

Comment:
Same question, are these features specified already?

Reply:
No, it's not. In the specification, we only tell to leave bit fields available for that. The actual implementation of EDC or ECC is not included in the OpenHW project.

Reply:
same suggestion as above

Reply:
I have implemented the same way.